src/drivers/cmakeDriver.ts

```diff
@@ -1742,12 +1820,33 @@ export abstract class CMakeDriver implements vscode.Disposable {
                 }
             }
             const useBuildTask: boolean = this.config.buildTask && isBuildCommand === true;
-            const exeOpt: proc.ExecutionOptions = { environment: buildcmd.build_env, outputEncoding: outputEnc, useBuildTask: useBuildTask };
-            const child = this.executeCommand(buildcmd.command, buildcmd.args, consumer, exeOpt);
-            this._currentBuildProcess = child;
-            await child.result;
-            this._currentBuildProcess = null;
-            return child;
+            if (useBuildTask) {
+                const task = await this._findBuildTask(targets);
+                if (task) {
+                    this._requestedTargets = targets;
+                    await vscode.tasks.executeTask(task);
+
+                    return { child: undefined, result: new Promise<api.ExecutionResult>(resolve => {
+                        const disposable = vscode.tasks.onDidEndTask(e => {
+                            if (e.execution.task.group?.id === vscode.TaskGroup.Build.id) {
+                                disposable.dispose();
+                                resolve({ retc: 0, stdout: '', stderr: '' });
+                            }
+                        });
+                    }) };
+                }
+                return { child: undefined, result: new Promise<api.ExecutionResult>((resolve) => {
+                    resolve({ retc: 0, stdout: '', stderr: '' });
```

Comment:
I think this should be an error and there should be come communication back to the developer that tells them we didn't find a task to build with.

Reply:
Agree. However I am not an expert (my domain is rather embedded C not typescript) so I used code which was used before while task was triggered (AFAIR it was execute methid in proc.ts).